lib/seattleflu/db/cli/command/etl/kit_result.py

```diff
@@ -0,0 +1,349 @@
+                    continue
+
+                kit = find_kit(db, kit_identifier.uuid)
+
+                # Error out the kit-result etl process if no kit found
+                # The kit-result etl process can try again starting with this
+                # record next time with the idea that the kit will be created
+                # by then.
+                if not kit:
+                    raise KitNotFoundError(f"No kit with «{kit_identifier.uuid}» found")
+
+                result_type = kit_result.document.pop("type")
+
+                if result_type == "strip":
+                    LOG.debug("Processing results for test strip")
+
+                    # Get test strip details from kit found in warehouse
+                    kit_test_strip = kit.details.get("test_strip")
+
+                    if not kit_test_strip:
+                        LOG.warning(f"Kit {kit.id} does not contain any test-strip data.")
+                        insert_kit_test_strip(db,
+                            kit                 = kit,
+                            test_strip_results  = kit_result.document,
+                            identifier_set      = "test-strips-fluathome.org")
+
+                    else:
+                        test_strip_barcode = kit_result.document.pop("barcode")
+
+                        if (test_strip_barcode and
+                            test_strip_barcode.lower() != kit_test_strip.get("barcode")):
+                            LOG.warning(f"Test strip barcode {test_strip_barcode} does not match " + \
+                                        f"barcode found in kit details: {kit_test_strip.get('barcode')}")
+
+                        # update kit details with test-strip results
+                        update_kit_test_strip(db, kit, kit_result.document)
+
+                else:
+                    LOG.debug("Processing results for kit sample")
+
+                    test_date = kit_result.document.pop("test_date", None)
+                    sample_barcode = kit_result.document.pop("barcode")
+                    sample_identifier = find_identifier(db, sample_barcode)
+
+                    if not sample_identifier:
+                        LOG.warning(f"Unknown {result_type} sample barcode {sample_barcode}")
+                        mark_skipped(db, kit_result.id)
+
+                    else:
+                        sample = find_sample(db, sample_identifier.uuid)
+
+                        assert sample.id in {kit.rdt_sample_id, kit.utm_sample_id}, \
+                            (f"Found sample {sample.id} does not match samples " +
+                             f"linked to kit: {kit.rdt_sample_id, kit.utm_sample_id}")
+
+                        # Loop through test results
+                        for test_result in kit_result.document:
+
+                            target_result = get_target_result(kit_result.document[test_result])
+
+                            if target_result is None:
```

Comment:
At this point we are now three conditionals deep. I think there may be a way to rewrite the code to reduce indentation a bit. 

For example, on line 123, after we mark the kit_result as skipped if it doesn't have a sample_identifier, then we can `continue` (like you use in lines 139-141).